### PR TITLE
Upgrade to `nightly-2022-11-16`

### PR DESCRIPTION
**Pull request by jdisanti**
This PR upgrades the nightly compiler that `cargo-check-external-types` is compatible with to `nightly-2022-11-16`.

_By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice._

### CODE REVIEW

**Comment by jdisanti** [comment]
`CI / Smoketest against the AWS Rust SDK` is failing due to ICE https://github.com/rust-lang/rust/issues/104145

File: src/error.rs [hunk 1]
```diff
@@ -94,6 +97,12 @@ impl ValidationError {
             "{}:{type_name}:{what}:{in_what_type}",
             location_sort_key(location)
         );
+        if location.is_none() {
+            eprintln!(
+                "WARN: An error is missing a span and will be printed without context, file name, and line number. \
+                This is a bug. Please report it with a piece of Rust code that triggers it at: {NEW_ISSUE_URL}"
```
Comment (Velfi) [inline src/error.rs:103]:
```suggestion
                This is a bug. Please report it with a [minimal reproducible example](https://en.wikipedia.org/wiki/Minimal_reproducible_example) that triggers it at: {NEW_ISSUE_URL}"
```

Comment (jdisanti) [reply src/error.rs]: It's very difficult to come up with one of these without sprinkling `cargo-check-external-types` with carefully placed panics that output the type ID that caused the issue, so I don't really want to require that. It's easy for me to do, but hard for someone not familiar with the tool's code to do.

File: src/error.rs [hunk 2]
```diff
@@ -112,48 +159,63 @@ impl ValidationError {
...
+                write!(
+                    f,
+                    "Fields on `{type_name}` marked `#[doc(hidden)]` cannot be checked for external types"
                 )
             }
         }
     }
 
-    pub fn subtext(&self) -> String {
+    pub fn subtext(&self) -> Cow<'static, str> {
```
Comment (Velfi) [inline src/error.rs:213]: https://www.youtube.com/watch?v=Yk7M2jGdnxU

**Review by Velfi — APPROVED** [review]

**Pull request merged**